ui/app/upload_image/controller.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const multer = require('multer')
+const httpRequest = require('request')
+const returnError = require('../errors/index')
+const logger = require('../logging/index')
+const uploadFile = multer({
+    destination: './',
+    limits: {
+        fileSize: 5000000 // limit file size to 5MB
+    }
+})
+
+// handles HTTP POST for the route /upload-image
+const handleUploadImageRoute = [
+    uploadFile.single('file'),
+    (request, response, next) => {
+        const token = request.session.jwt
+        const file = request.file
+        const userId = request.session.userId
+        handleUploadingImage(file, userId, token).then((image) => {
+            if (image['sloth_check'].contains_sloth) response.redirect('/image-library')
+            else {
+                response.status(200).render('index', {page: 'Slothbucket', contains_sloth: 'There\'s no sloth!'})
+            }
+        }).catch((error) => {
+            if (error.code && error.code === 400) {
+                response.status(200).render('index', {page: 'Slothbucket', errors: [{msg: error.message}]})
+            } else next(error)
+        })
+    }
+]
+
+/**
+ * Sends a HTTP POST to the backend's route /classify-image
+ * @param base64 {string} - base64 image
+ * @param userId {string} - user's id
+ * @param token {string} - JWT of authenticated user
+ * @returns {Promise} - Promise representing if posting to the route was successful
+ */
+function postClassifyImage(base64, userId, token) {
+    return new Promise((resolve, reject) => {
+        const options = {
+            url: 'http://localhost:8000/classify-image',
```

Comment:
For the URLs, it would be better to move it out to some config (file or environment variable). Once this is deployed, it's not a guarantee that the backend is broadcasting on localhost or port 8000